handlers/root.go

```diff
@@ -13,3 +16,17 @@ func Root(w http.ResponseWriter, r *http.Request) {
 func getStatus(w http.ResponseWriter, r *http.Request) {
 	RespondWithJSONIndent(w, http.StatusOK, status.Get())
 }
+
+func Auth(w http.ResponseWriter, r *http.Request) {
+	u, p, ok := r.BasicAuth()
+	if !ok {
+		RespondWithJSONIndent(w, http.StatusInternalServerError, nil)
+		return
+	}
+	token, err := security.NewAuthToken(u+p, time.Now().Local().Add(time.Hour*time.Duration(config.Get().Token.ExpirationHours)), config.Get().Token.Secret)
```

Comment:
Not sure why my previous comment went outdated, I was saying:

`r.BasicAuth` doesn't actually check anything. I mean, we still need to make sure it matches somehow the username and password we have in our config file, conf.Server.Credentials.Username and conf.Server.Credentials.Password . Else I bet you can login with any credential :)

Reply:
I don't know If I am udnerstanding you.
In server.go L-87 is checking that the username and password are the same

```go
else if h.credentials.Username != "" || h.credentials.Password != "" {
		u, p, ok := r.BasicAuth()
		if !ok || h.credentials.Username != u || h.credentials.Password != p {
			statusCode = http.StatusUnauthorized
		}
```

h is called in the handler where proxyHandler have the config file credentials

```go
proxyHandler := serverAuthProxyHandler{
		credentials: security.Credentials{
			Username: conf.Server.Credentials.Username,
			Password: conf.Server.Credentials.Password,
		},
		trueHandler: router,
	}
	http.HandleFunc("/", proxyHandler.handler)
```

So if a try to access with another credentials fail

![screenshot from 2018-05-28 13-44-36](https://user-images.githubusercontent.com/3019213/40613071-47fc2040-627d-11e8-829b-41e350d4d70a.png)

Reply:
Ok my bad, I didn't see the whole picture. If the authorization handler is still called in addition to the new one then I suppose it's fine! :+1: